gateway/sds_gateway/users/views.py

```diff
@@ -1806,3 +1815,567 @@ def post(
+                        }
+                    )
+                else:
+                    error_msg = f"Failed to upload {filename}: {response.data}"
+                    errors.append(error_msg)
+                    logger.error(error_msg)
+            errors.extend(upload_errors)
+        errors.extend(skipped_files)
+        return saved_files, errors
+
+    def _create_single_capture(
+        self,
+        request,
+        channel,
+        top_level_dir,
+        capture_type,
+        scan_group,
+        all_relative_paths=None,
+    ):
+        """Create a single capture for a channel.
+
+        Returns (capture_data, error) where capture_data is the created capture
+        data or None if creation failed, and error is the error message or None.
+        """
+        try:
+            # Determine capture type
+            if capture_type == "rh":
+                capture_type_enum = CaptureType.RadioHound
+            else:
+                capture_type_enum = CaptureType.DigitalRF
+
+            # Prepare capture data for the helper function
+            capture_data = {
+                "capture_type": capture_type_enum,
+                "top_level_dir": str(top_level_dir),
+            }
+
+            # Add appropriate field based on capture type
+            if capture_type == "rh":
+                capture_data["scan_group"] = scan_group
+            else:
+                capture_data["channel"] = channel
+
+            # Set the index name based on capture type
+            from sds_gateway.api_methods.utils.metadata_schemas import infer_index_name
+
+            capture_data["index_name"] = infer_index_name(capture_type_enum)
+
+            # Use the helper function to create the capture
+            responses, capture_errors = create_capture_helper_simple(
+                request, capture_data
+            )
+
+            if responses:
+                # Capture created successfully
+                response = responses[0]
+                if hasattr(response, "data") and isinstance(response.data, dict):
+                    capture_data = response.data
+                    return capture_data, None
+                logger.warning(
+                    "Unexpected response format for channel %s: %s",
+                    channel,
+                    response.data,
+                )
+                return (
+                    None,
+                    f"Unexpected response format for channel {channel}: "
+                    f"{response.data}",
+                )
+            # Capture creation failed
+            error_msg = capture_errors[0] if capture_errors else "Unknown error"
+            logger.error(
+                "Failed to create capture for channel %s: %s",
+                channel,
+                error_msg,
+            )
+            return (  # noqa: TRY300
+                None,
+                f"Failed to create capture for channel {channel}: {error_msg}",
+            )
```

Comment:
return out of the try block

https://docs.astral.sh/ruff/rules/try-consider-else/